src/databases/Mili/avtMiliFileFormat.C

```diff
@@ -74,4384 +53,3837 @@ extern "C" {
+                int *part       = new int[nCells];
+                char *shortName = new char[classNames[cpcIdx].size() + 1];
+                strcpy(shortName, classNames[cpcIdx].c_str());
+                cpcIdx++;
+
+                int rval = mc_load_conns(dbid[dom], meshId, 
+                    shortName, conn, mats, part);
+
+                if (rval != OK)
+                {
+                    EXCEPTION1(InvalidVariableException, shortName);
+                }
+
+                //
+                // Create a conn pointer that we can move around
+                // in memory without offsetting the original.
+                //
+                int *connPtr = conn;
+    
+                for (int k = 0 ; k < nCells ; ++k)
                 {
+                    matList[cellIdx] = mats[k];
+                    cellIdx++;
+
                     vtkIdType verts[100];
-                    for(int cc = 0; cc < conn_count[i]; ++cc)
-                        verts[cc] = (vtkIdType)conn[cc];
+
+                    for(int cc = 0; cc < connCount; ++cc)
+                    {
+                        verts[cc] = (vtkIdType)connPtr[cc];
+                    }
                     
-                    switch (elem_sclasses[i])
+                    switch (miliCellTypes[i])
                     {
-                      case M_TRUSS:
-                        connectivity[dom][mesh_id]->InsertNextCell(VTK_LINE,
-                                                 conn_count[i], verts);
-                        break;
-                      case M_BEAM:
-                        // Beams are lines that have a third point to define
-                        // the normal. Since we don't need to visualize it,
-                        // we just drop the normal point.
-                        connectivity[dom][mesh_id]->InsertNextCell(VTK_LINE,
-                                                 2, verts);
-                        break;
-                      case M_TRI:
-                        connectivity[dom][mesh_id]->InsertNextCell(
-                                                                  VTK_TRIANGLE,
-                                                 conn_count[i], verts);
-                        break;
-                      case M_QUAD:
-                        connectivity[dom][mesh_id]->InsertNextCell(VTK_QUAD,
-                                                     conn_count[i], verts);
-                        break;
-                      case M_TET:
-                        connectivity[dom][mesh_id]->InsertNextCell(VTK_TETRA,
-                                                     conn_count[i], verts);
-                        break;
-                      case M_PYRAMID:
-                        connectivity[dom][mesh_id]->InsertNextCell(VTK_PYRAMID,
-                                                     conn_count[i], verts);
-                        break;
-                      case M_WEDGE:
-   
```

Comment:
Any insights with the mem errors here?

Reply:
Just figured this out. Thanks for getting me to re-visit this!